File: app/src/pfp/java/social/entourage/android/navigation/BottomNavigationDataSource.java
package social.entourage.android.navigation;

import social.entourage.android.R;
import social.entourage.android.map.MapEntourageFragment;
import social.entourage.android.entourage.my.MyEntouragesFragment;
import social.entourage.android.map.MapVoisinage;
import social.entourage.android.sidemenu.SideMenuFragment;

/**
 * Created by Mihai Ionescu on 23/04/2018.
 */
public class BottomNavigationDataSource extends BaseBottomNavigationDataSource {

    public BottomNavigationDataSource() {
        add(new NavigationItem(R.string.action_map, R.drawable.ic_navigation_map), new MapVoisinage(), MapVoisinage.TAG);
        add(new NavigationItem(R.string.action_my_messages, R.drawable.ic_navigation_my_messages), new MyEntouragesFragment(), MyEntouragesFragment.TAG);
        add(new NavigationItem(R.string.action_menu, R.drawable.ic_navigation_menu), new SideMenuFragment(), SideMenuFragment.TAG);

        defaultSelectedTab = 1; // My messages tab
    }

}

File: app/src/pfp/java/social/entourage/android/navigation/BottomNavigationDataSource.kt
package social.entourage.android.navigation

import androidx.fragment.app.Fragment
import social.entourage.android.R
import social.entourage.android.entourage.my.MyEntouragesFragment
import social.entourage.android.newsfeed.BaseNewsfeedFragment
import social.entourage.android.map.MapVoisinageFragment
import social.entourage.android.mainprofile.MainProfileFragment

/**
 * Created by Mihai Ionescu on 23/04/2018.
 */
class BottomNavigationDataSource : BaseBottomNavigationDataSource() {
    override fun getFragmentAtIndex(menuId: Int): Fragment? {
        return when (menuId) {
            R.id.bottom_bar_newsfeed -> MapVoisinageFragment()
            R.id.bottom_bar_mymessages -> MyEntouragesFragment()
            R.id.bottom_bar_profile -> MainProfileFragment()
            else -> null
        }
    }

    init {
        add(R.id.bottom_bar_newsfeed, BaseNewsfeedFragment.TAG)
        add(R.id.bottom_bar_mymessages, MyEntouragesFragment.TAG)
        add(R.id.bottom_bar_profile, MainProfileFragment.TAG)
        defaultSelectedTab = R.id.bottom_bar_mymessages // My messages tab
    }
}